src/Desktop.Analyzers/Core/DoNotUseInsecureDTDProcessing.cs

```diff
@@ -137,215 +176,135 @@ public void AnalyzeCodeBlockEnd(CodeBlockAnalysisContext context)
-                                else
-                                {
-                                    isXmlTextReaderDtdProcessingDisabled = true;
-                                }
-                            }
-                        }
-                        if (!isXmlTextReaderSecureResolver || !isXmlTextReaderDtdProcessingDisabled)
-                        {
-                            Diagnostic diag = Diagnostic.Create(
+                        DiagnosticDescriptor rule = RuleDoNotUseInsecureDTDProcessing;
+                        Diagnostic diag = Diagnostic.Create(
                                 RuleDoNotUseInsecureDTDProcessing,
-                                node.GetLocation(),
+                                location,
                                 SecurityDiagnosticHelpers.GetLocalizableResourceString(
-                                    nameof(DesktopAnalyzersResources.XmlTextReaderConstructedWithNoSecureResolutionMessage),
-                                    SecurityDiagnosticHelpers.GetNonEmptyParentName(node, model)
+                                    nameof(DesktopAnalyzersResources.XmlReaderCreateWrongOverloadMessage)
                                 )
                             );
-                            context.ReportDiagnostic(diag);
-                        }
-                    }
-                }
-                else if (method.MatchMethodDerivedByName(xmlTypes.XmlReader, SecurityMemberNames.Create))
-                {
-                    int xmlReaderSettingsIndex = SecurityDiagnosticHelpers.HasXmlReaderSettingsParameter(method, xmlTypes);
-                    if (xmlReaderSettingsIndex < 0)     //FxCop CA3053:XmlReaderCreateWrongOverload
-                    {
-                        DiagnosticDescriptor rule = RuleDoNotUseInsecureDTDProcessing;
-                        context.ReportDiagnostic(
-                            Diagnostic.Create(
-                                rule,
-                                node.GetLocation(),
-                                SecurityDiagnosticHelpers.GetLocalizableResourceString(
-                                    nameof(DesktopAnalyzersResources.XmlReaderCreateWrongOverloadMessage),
-                                    SecurityDiagnosticHelpers.GetNonEmptyParentName(node, model)
-                                )
-                            )
-                        );
+                        context.ReportDiagnostic(diag);
                     }
                     else
                     {
-                        SyntaxNode settingsNode = _syntaxNodeHelper.GetInvocationArgumentExpressionNodes(node).ElementAt(xmlReaderSettingsIndex);
-                        ISymbol settingsSymbol = SyntaxNodeHelper.GetSymbol(settingsNode, model);
+
+                        IArgument arg = arguments[xmlReaderSettingsIndex];
+                        ISymbol settingsSymbol = arg.Value.Syntax.GetDeclaredOrReferencedSymbol(model);
```

Comment:
The intention of this extension method is to find what symbol is being declared or referenced within this SyntaxNode node. model is only a helper class here to retrieve symbol information. So  with that said, it is more specific to this node, instead of the entire model. When we look at where a extension method belongs to, we should look at the intention of this method instead of implementation. So I think it is better to stay with SyntaxNode..